generator/metaline_builder.cpp

```diff
@@ -0,0 +1,217 @@
+#include "generator/metaline_builder.hpp"
+#include "generator/routing_helpers.hpp"
+
+#include "indexer/classificator.hpp"
+
+#include "coding/file_container.hpp"
+#include "coding/varint.hpp"
+#include "coding/write_to_sink.hpp"
+
+#include "base/checked_cast.hpp"
+#include "base/logging.hpp"
+#include "base/string_utils.hpp"
+
+#include "defines.hpp"
+
+#include <cstdint>
+
+namespace
+{
+using Ways = std::vector<std::int32_t>;
+
+class LineString
+{
+  Ways m_ways;
+  std::uint64_t m_start;
+  std::uint64_t m_end;
+  bool m_oneway;
+
+public:
+  explicit LineString(OsmElement const & way)
+  {
+    std::string const oneway = way.GetTag("oneway");
+    m_oneway = !oneway.empty() && oneway != "no";
+    std::int32_t const wayId = base::checked_cast<std::int32_t>(way.id);
+    m_ways.push_back(oneway == "-1" ? -wayId : wayId);
+    m_start = way.Nodes().front();
```

Comment:
This is actually impossible due to the calling method in `osm_translator`, but whatever.